File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* cur1 = list1;
        ListNode* cur2 = list2;
        ListNode* dummy =  new ListNode();
        ListNode* cur3 = dummy;
        while(cur1 && cur2)
        {
           if(cur1->val<=cur2->val){
             cur3->next = cur1;
             cur1 = cur1->next;
           }
           else{
            cur3->next = cur2;
            cur2 = cur2->next;
           }
           cur3 = cur3->next;
        }
        if(cur1)
            cur3->next = cur1;
        if(cur2)
            cur3->next = cur2;
        return dummy->next;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        
        h1 = list1 
        h2 = list2
        dummy = ListNode()
        c = dummy
        while(h1 and h2):
            if h1.val<=h2.val:
                temp = ListNode(h1.val)
                h1 = h1.next
            else :
                temp = ListNode(h2.val)
                h2 = h2.next
            c.next = temp
            c = c.next

        if h1:
            c.next = h1  
        elif h2:
            c.next = h2
        return dummy.next                  
                



        